scripts/blant-clusters.sh

```diff
@@ -212,31 +213,53 @@ for k in "${Ks[@]}"; do
 	sort -nr | # sort the above output by number of nodes in the near-clique
 	hawk 'BEGIN{ numCliques=0 } # post-process to remove duplicates
 	    {
-		delete S; seenColon=0;
+		delete S;
 		numNodes=$1
 		edgeHits=$2;
 		for(i=3;i<=NF;i++) ++S[$i]
 		ASSERT(length(S)==numNodes,"mismatch in numNodes and length(S)");
 		    for(i=1;i<=numCliques;i++) {
 			    same=0;
 			    for(u in S) if(u in cluster[i])++same;
-			    if(same == length(S)) break;
+			    if(same >= length(S)*'$t') break;
 		}
-		if(numCliques==0 || same < length(S)) {
+		if(numCliques==0 || same < length(S)*'$t') {
 			    maxEdges=choose(length(S),2);
-			    ++numCliques; printf "%d nodes, %d of %d edges from k '$k' (%g%%):",
-			length(S), edgeHits, maxEdges, 100*edgeHits/maxEdges
+			    ++numCliques; 
+				printf "%d %d '$k'",length(S),edgeHits
 			    for(u in S) {cluster[numCliques][u]=1; printf " %s", u}
 			    print ""
 		}
-	    }' | sort -k 1nr -k 11n > $TMPDIR/final$k.out & # sort by number of nodes and then by the first node in the list
+	    }' | sort -k 1nr -k 4n > $TMPDIR/final$k.out & # sort by number of nodes and then by the first node in the list
 done
 
 for k in "${Ks[@]}"; do
     wait; (( BLANT_EXIT_CODE += $? ))
 done
 
-sort -k 1nr -k 11n $TMPDIR/final?.out
+sort -k 1nr -k 3n $TMPDIR/final?.out |
+hawk 'BEGIN{ numCliques=0 } # post-process to remove duplicates
+	    {
+		delete S; 
+		numNodes=$1
+		edgeHits=$2;
+		k=$3;
+		for(i=4;i<=NF;i++) ++S[$i]
+		ASSERT(length(S)==numNodes,"mismatch in numNodes and length(S)");
+		for(i=1;i<=numCliques;i++) {
+			    same=0;
+			    for(u in S) if(u in cluster[i])++same;
+			    if(same >= length(S)*'$t') break;
+		}
+		if(numCliques==0 || same < length(S)*'$t') {
+			    maxEdges=choose(length(S),2);
+			    ++numCliques; 
+				printf "%d nodes, %d of %d edges from k %d (%g%%):",
+			length(S), edgeHits, maxEdges, k, 100*edgeHits/maxEdges
+			    for(u in S) {cluster[numCliques][u]=1; printf " %s", u}
+			    print ""
+		}
+	    }' - | sort -k 1nr -k 11n # sort by number of nodes and then by the first node in the list
```

Comment:
I think the second sort should be changed to be similar to the one above, to sort by column 3 rather than 11??